server/routes/mail.coffee

```diff
@@ -742,7 +742,7 @@ module.exports.sendNextStepsEmail = sendNextStepsEmail = (user, now, daysAgo) ->
           secretLevelName: secretLevel.name
           secretLevelLink: "http://codecombat.com/play/level/#{secretLevel.slug}"
           levelsComplete: complete.length
-          isCoursePlayer: user.get('courseInstances')?.length > 0
+          isCoursePlayer: user.get('courseInstances')?.length > 0 # TODO: use based on role instead, as courseInstances can be unreliable
```

Comment:
Next time this area is worked on. The TODO's purpose is to avoid one bug fix causing a long cascade of fixes, but not entirely lose knowledge gathered.